runtime/lua/vim/lsp/completion.lua

```diff
@@ -257,7 +257,13 @@ local function match_item_by_value(value, prefix)
     return true
   end
   if vim.o.completeopt:find('fuzzy') ~= nil then
-    return next(vim.fn.matchfuzzy({ value }, prefix)) ~= nil
+    local fuzzyopts = vim.tbl_get(buf_handles, api.nvim_get_current_buf(), 'fuzzyopts')
+    if fuzzyopts and next(fuzzyopts) == nil then
+      fuzzyopts = nil
+    end
+    local result = fuzzyopts and vim.fn.matchfuzzy({ value }, prefix, fuzzyopts)
+      or vim.fn.matchfuzzy({ value }, prefix)
```

Comment:
If people want to screw around with the matching then they are going to ask for all kinds of stuff. Adding "fuzzyopts" is obviously too narrow. The usual way we approach this is to add some kind of `filter` fn.

Something like this (just a sketch):

```suggestion
    local result = filterfn and filterfn({ value }, prefix) or vim.fn.matchfuzzy({ value }, prefix)
```

Reply:
Good idea. I was kind of hoping that starting from any idea would revive the discussion on what it should ultimately look like.